OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -62,6 +62,19 @@ public class ArmamentInfo : UpgradableTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		[Desc("Is it possible to use the armament when force-attacking?")]
+		public readonly bool ForceAttacks = true;
+
+		[Desc("Is the armament beneficial (should it be used on allies)?")]
+		public readonly bool Beneficial = false;
```

Comment:
This is the only place where `Beneficial` is used

https://github.com/OpenRA/OpenRA/pull/8768/files#diff-26d2cbcfb01852edd1172678f8116784R244

Right now,
* if an armament has `Beneficial == false`,
  * it can be used to target enemies as usual
    * without force-attack,
    * and with it if additionally `ForceAttacks == true`
  * and allies
    * only using force-attack if additionally `ForceAttacks == true`
* if `Beneficial == true`,
  * only allies can be targeted with it
    * without force-attack,
    * and with it if additionally `ForceAttacks == true`
  * enemies cannot be targeted at all
    * neither with
    * nor without force-attack